### PR TITLE
task/INBA-529-PO-Add Message Services (notify) call from Discussions and Tasks

### PR DESCRIPTION
**Pull request by Perry5**
Add ability for users to get internal messages as well as emails when they get flagged. 

To test:
- Go through steps to create project, assign task etc.
- Comment on a question and notify user.
- Log in as that user and make sure they get notification on Indaba and they also get an email.

### CODE REVIEW

File: backend/app/controllers/notifications.js [hunk 1]
```diff
@@ -820,7 +823,8 @@ function* extendNote(req, note, userTo, essenceName, entityId, orgId, taskId) {
     var product = yield * common.getEntity(req, task.productId, Product, 'id');
     var uoa = yield * common.getEntity(req, task.uoaId, UOA, 'id');
     var step = yield * common.getEntity(req, task.stepId, WorkflowStep, 'id');
-    var survey = yield * common.getEntity(req, product.surveyId, Survey, 'id');
+    // var survey = yield * common.getEntity(req, product.surveyId, Survey, 'id'); // TODO: Pull Survey here
```
Comment (Perry5) [inline backend/app/controllers/notifications.js]: I will add call here to pull survey from survey service.

File: backend/app/controllers/notifications.js [hunk 2]
```diff
@@ -829,8 +833,8 @@ function* extendNote(req, note, userTo, essenceName, entityId, orgId, taskId) {
         product: product,
         uoa: uoa,
         step: step,
-        survey: survey,
-        policy: survey,
+        // survey: survey,
+        // policy: survey,
```
Comment (Perry5) [inline backend/app/controllers/notifications.js]: When survey is pulled, it'll go here.

**Comment by Perry5** [comment]
Updated this with Survey from survey service!

File: backend/app/controllers/system_messages.js [hunk 3]
```diff
@@ -9,49 +8,14 @@ module.exports = {
...
             res.status(400).send('System message must have a \'to\' field and a \'message\' field').end();
         } else {
-            messageService.sendSystemMessage(
-                req.app.get(messageService.SYSTEM_MESSAGE_USER_TOKEN_FIELD),
-                req.body.to,
-                req.body.message,
-                messageService.SYSTEM_MESSAGE_SUBJECT
-            )
-            .then(() => {
+            const response = yield * common.sendSystemMessageWithMessageService(req, req.body.to, req.body.message);
```
Comment (sirctseb) [inline backend/app/controllers/system_messages.js]: This `yield` is undefined, giving a 400 `{
  "message": "yield is not defined"
}` in response to the endpoint. I think we need a `*` on this function declaration

File: backend/app/services/common.js [hunk 4]
```diff
@@ -656,6 +656,61 @@ var bumpProjectLastUpdated = function *(req, projectId) {
...
+                })
+                .then(() =>
+                    messageService.sendSystemMessage(
+                        req.app.get(messageService.SYSTEM_MESSAGE_USER_TOKEN_FIELD),
+                        to,
+                        message,
+                        messageService.SYSTEM_MESSAGE_SUBJECT
+                    )
+                    .then((res) => {
+                        console.log(`SUCCEEDED THIS TIME - RETURNING RES: ${res.statusCode}`)
```
[the remark this answers is not in the document]
Comment (Perry5) [reply backend/app/services/common.js]: Done

**Comment by Perry5** [comment]
Updated!

**Comment by Perry5** [comment]
6fb7db0a04548851963e3275ebfb89ec

Merging!

**Pull request merged**